app/lib/common/widgets/html_editor/components/mention_list.dart

```diff
@@ -0,0 +1,289 @@
+                  mentionType: widget.mentionType,
+                  displayName: displayName,
+                  avatarOptions: options,
+                  isSelected: index == _selectedIndex,
+                  onTap: () => _selectItem(mentionId, displayName),
+                );
+              },
+            ),
+    );
+  }
+
+  KeyEventResult _handleKeyEvent(
+    FocusNode node,
+    KeyEvent event,
+    Map<String, String> suggestions,
+  ) {
+    if (event is! KeyDownEvent) return KeyEventResult.ignored;
+
+    switch (event.logicalKey) {
+      case LogicalKeyboardKey.escape:
+        widget.onDismiss();
+        return KeyEventResult.handled;
+
+      case LogicalKeyboardKey.enter:
+        if (suggestions.isNotEmpty) {
+          final selectedItem = suggestions.entries.elementAt(_selectedIndex);
+          _selectItem(selectedItem.key, selectedItem.value);
+        }
+        widget.onDismiss();
+        return KeyEventResult.handled;
+
+      case LogicalKeyboardKey.arrowUp:
+        setState(() {
+          _selectedIndex =
+              (_selectedIndex - 1).clamp(0, suggestions.length - 1);
+        });
+        _scrollToSelected();
+        return KeyEventResult.handled;
+
+      case LogicalKeyboardKey.arrowDown:
+        setState(() {
+          _selectedIndex =
+              (_selectedIndex + 1).clamp(0, suggestions.length - 1);
+        });
+        _scrollToSelected();
+        return KeyEventResult.handled;
+
+      case LogicalKeyboardKey.backspace:
+        final selection = widget.editorState.selection;
+        if (selection == null) return KeyEventResult.handled;
+
+        final node = widget.editorState.getNodeAtPath(selection.end.path);
+        if (node == null) return KeyEventResult.handled;
+
+        // Get text before cursor
+        final text = node.delta?.toPlainText() ?? '';
+        final cursorPosition = selection.end.offset;
+
+        if (_canDeleteLastCharacter()) {
+          // Check if we're about to delete an mention symbol
+          if (cursorPosition > 0 &&
+              text[cursorPosition - 1] ==
+                  MentionType.toStr(widget.mentionType)) {
+            widget.onDismiss(); // Dismiss menu when is deleted
+          }
+          widget.editorState.deleteBackward();
+        } else {
+          // Workaround for editor regaining focus
+          widget.editorState.apply(
+            widget.editorState.transaction..afterSelection = selection,
+          );
+        }
+        return KeyEventResult.handled;
+
+      default:
+        if (event.character != null &&
+                !HardwareKeyboard.instance.isAltPressed &&
+                !HardwareKeyboard.instance.isMetaPressed ||
+            !HardwareKeyboard.instance.isShiftPressed) {
+          widget.editorState.insertTextAtCurrentSelection(event.character!);
+          return KeyEventResult.handled;
+        }
+        return KeyEventResult.ignored;
+    }
+  }
+
+  void _selectItem(String id, String displayName) {
```

Comment:
handler when user selects mention item from list and processes how the editor should execute mention block inline